### PR TITLE
Feat: add import vendor packages.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

#223 

#### 2. What is the scope of this PR (e.g. component or file name):

kclvm-parser

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

In the research of the import path, the search for the external package is added. 
If  the pakcage can not be found in current package, the search path of the external package specified by the environment variable VENDOR_PATH will be searched.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: kclvm/tools/src/vet/tests.rs [hunk 1]
```diff
@@ -457,7 +457,7 @@ mod test_validater {
 
                 let result = panic::catch_unwind(|| validate(opt));
 
-                let mut expect: PanicInfo = serde_json::from_str(&expected_err_msg).unwrap();
+                let expect: PanicInfo = serde_json::from_str(&expected_err_msg).unwrap();
```
Comment (Peefy) [inline kclvm/tools/src/vet/tests.rs:460]: `let mut expect` is only for windows

File: kclvm/config/src/modfile.rs [hunk 2]
```diff
@@ -2,13 +2,31 @@
...
 use serde::Deserialize;
-use std::io::Read;
+use std::{env, io::Read};
 use toml;
 
 pub const KCL_MOD_FILE: &str = "kcl.mod";
 pub const KCL_FILE_SUFFIX: &str = ".k";
 pub const KCL_FILE_EXTENSION: &str = "k";
 pub const KCL_MOD_PATH_ENV: &str = "${KCL_MOD}";
+pub const VENDOR_HOME: &str = "VENDOR_HOME";
```
Comment (Peefy) [inline kclvm/config/src/modfile.rs:12]: Use `KCLVM_VENDOR_HOME` to prevent conflicts like `RUSTC_XX`?

File: kclvm/tools/src/vet/tests.rs (hunk 1 above)
Comment (zong-zhe) [reply kclvm/tools/src/vet/tests.rs]: fixed.

File: kclvm/config/src/modfile.rs (hunk 2 above)
Comment (zong-zhe) [reply kclvm/config/src/modfile.rs]: fixed.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**